src/preset.ts

```diff
@@ -1020,119 +1148,54 @@ async function tryApplyVsDevEnv(preset: ConfigurePreset) {
+                        silent: true,
+                        encoding: 'utf8',
+                        shell: true
+                    }).result;
 
-                        if (!vsInstall) {
-                            log.error(localize('specified.cl.not.found',
-                                "Configure preset {0}: Compiler {1} with toolset {2} and architecture {3} was not found, you may need to run the 'CMake: Scan for Compilers' command if this toolset exists on your computer.",
-                                preset.name, `"${compilerName}.exe"`, toolset.version ? `"${toolset.version},${toolset.host}"` : `"${toolset.host}"`, `"${arch}"`));
-                        } else {
-                            log.info(localize('using.vs.instance', "Using developer environment from Visual Studio (instance {0}, version {1}, installed at {2})", vsInstall.instanceId, vsInstall.installationVersion, `"${vsInstall.installationPath}"`));
-                            const vsEnv = await varsForVSInstallation(vsInstall, toolset.host!, arch, toolset.version);
-                            compilerEnv = vsEnv ?? EnvironmentUtils.create();
-
-                            // if ninja isn't on path, try to look for it in a VS install
-                            const ninjaLoc = await execute(whereExecutable, ['ninja'], null, {
-                                environment: EnvironmentUtils.create(preset.environment),
-                                silent: true,
-                                encoding: 'utf8',
-                                shell: true
-                            }).result;
-                            if (!ninjaLoc.stdout) {
-                                const vsCMakePaths = await paths.vsCMakePaths(vsInstall.instanceId);
-                                if (vsCMakePaths.ninja) {
-                                    log.warning(localize('ninja.not.set', 'Ninja is not set on PATH, trying to use {0}', vsCMakePaths.ninja));
-                                    compilerEnv['PATH'] = `${path.dirname(vsCMakePaths.ninja)};${compilerEnv['PATH']}`;
-                                }
-                            }
+                    const generatorIsNinja = preset.generator?.toLowerCase().includes("ninja");
+                    const shouldInterrogateForNinja = (generatorIsNinja ?? false) && !ninjaLoc.stdout;
 
-                            preset.environment = EnvironmentUtils.mergePreserveNull([preset.environment, compilerEnv]);
-                        }
+                    if (!compilerLocation.stdout || shouldInterrogateForNinja) {
+                        developerEnvironment = await getVsDevEnv({
+                            preset,
+                            shouldInterrogateForNinja,
+                            compilerName
+                        });
                     }
                 }
             }
+        } else if (useVsDeveloperEnvironmentMode === "always") {
```

Comment:
what happens if user has this set to always but wants to use a different ninja  than the one is vs dev environment that they prepended to their path in environment field? does this scenario work, or we always use vs devenv ninja?

Reply:
If they already prepended to their path in the environment field, then we will find it in our `where.exe` invocation, and therefore won't add the ninja PATH.